Google.PowerShell/BigQuery/BqTable.cs

```diff
@@ -330,4 +330,163 @@ public TablesResource.InsertRequest makeInsertReq()
+        /// The Dataset that you would like to search. This field takes Dataset or DatasetRefrence objects.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.ByDatasetObject)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Dataset),
+            Property = nameof(Apis.Bigquery.v2.Data.Dataset.DatasetReference))]
+        public DatasetReference Dataset { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The Table object that will be sent to the server to be inserted. Also takes TableReferences.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParameterSetNames.ByObject)]
+        [PropertyByTypeTransformation(TypeToTransform = typeof(Table),
+            Property = nameof(Apis.Bigquery.v2.Data.Table.TableReference))]
+        public TableReference InputObject { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Forces deletion of non-empty tables and the data contained in them.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        public SwitchParameter Force { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.ByValue:
+                    // No processing needed for ByValue parameter sets.
+                    break;
+                case ParameterSetNames.ByObject:
+                    Project = InputObject.ProjectId;
+                    DatasetId = InputObject.DatasetId;
+                    Table = InputObject.TableId;
+                    break;
+                case ParameterSetNames.ByDatasetObject:
+                    Project = Dataset.ProjectId;
+                    DatasetId = Dataset.DatasetId;
+                    break;
+                default:
+                    throw UnknownParameterSetException;
+            }
+
+            TablesResource.DeleteRequest request = Service.Tables.Delete(Project, DatasetId, Table);
+            String response = "Table Removal Stopped.";
+            if (ShouldProcess(Table))
+            {
+                try
+                {
+                    if (Force == true)
+                    {
+                        response = request.Execute();
+                    }
+                    else
+                    {
+                        TablesResource.GetRequest checkTables =
+                            new TablesResource.GetRequest(Service, Project, DatasetId, Table);
+                        var tableResponse = checkTables.Execute();
+                        if (tableResponse.NumRows == 0)
+                        {
+                            response = request.Execute();
+                        }
+                        else if (ShouldContinue(
```

Comment:
You may want to use the 4 parameter version of ShouldContinue. It may look like it is only called once, but if you pipeline in a bunch of non-empty table objects, a Yes to All/No to All option would be nice.